### PR TITLE
2023 03 release notes

### PR DESCRIPTION
**Pull request by prestwick**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/install-systemlink-enterprise/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Create release notes for 2023-03
- Create a release note template
- Create bug fix list

### Why should this Pull Request be merged?

2023-03 release

### What testing has been done?

NA

### CODE REVIEW

File: release-notes/2023-03/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        - You are no longer required to also set options in `kafka.metricsConfigs`, `kafka.jmxOptions`, and `kafka.kafkaExporter` configurations
+        - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-03/getting-started/templates/systemlink-values.yaml#L56)
+
+    - Enabling user claims to be returned in a web browser
+        - Set `webserver.oidc.enableUserInfo` to `true`
+        - You should remove `offline_access` from `webserver.oidc.scope`.
+        - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-03/getting-started/templates/systemlink-values.yaml#L110)
+
+    - Enabling Kafka UI
+        - To enable the Kafka UI you must override the image pull repository to the repository you are installing SystemLink Enterprise from.
```
Comment (cameronwaterman) [inline release-notes/2023-03/README.md:38]: They will also need to override the image pull secrets value. I've updated the helm breaking changes spreadsheet with details.

**Review by cameronwaterman — APPROVED** [review]

File: release-notes/2023-03/README.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+- Deleted Kafka S3 sink connectors are automatically reestablished if connectors are manually deleted.
+
+- The default memory provided to Kafka topic operator has increased from 256MB to 512MB to increase the total number of tables that can be written to by the Data Frame Service.
+
+## Helm Chart Breaking Changes
+
+- `systemlink 0.11.81`
+    - Enabling Kafka OpenTelemetry metrics
+        - Set `global.telemetry.enabled` flag to `true`.
+        - You are no longer required to also set options in `kafka.metricsConfigs`, `kafka.jmxOptions`, and `kafka.kafkaExporter` configurations.
```
Comment (prestwick) [inline release-notes/2023-03/README.md:35]: @cameronwaterman can you check the wording of this line? Its deliberately different from what's in the breaking changes spreadsheet due to a conversation we had, but I want to be sure I didn't misunderstand you.

File: release-notes/2023-03/README.md (hunk 1 above)
Comment (prestwick) [reply release-notes/2023-03/README.md:49]: @cameronwaterman this should be resolved, but please review the comments on line 33 of the release notes

File: release-notes/2023-03/README.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+- Deleted Kafka S3 sink connectors are automatically reestablished if connectors are manually deleted.
+
+- The default memory provided to Kafka topic operator has increased from 256MB to 512MB to increase the total number of tables that can be written to by the Data Frame Service.
+
+## Helm Chart Breaking Changes
+
+- `systemlink 0.11.81`
+    - Setting image pull secrets
+        - Image pull secret specification has been split into two separate global values.
+        - You must specifying the single required image pull secret in a new value `global.niImagePullSecret` and reference this value in `global.imagePullSecrets`.
```
Comment (cameronwaterman) [inline release-notes/2023-03/README.md:35]:
```suggestion
        - You must specify the single required image pull secret in a new value `global.niImagePullSecret` and reference this value in `global.imagePullSecrets`.
```

Comment (cameronwaterman) [reply release-notes/2023-03/README.md:35]: I think this wording is fine as long as consumers of this document understand to reference the template values file. The use of yaml anchors to achieve what this line is describing isn't exactly easy to put into words.

File: release-notes/2023-03/README.md (hunk 2 above)
Comment (cameronwaterman) [reply release-notes/2023-03/README.md:40]: This should be fine. It should be self-explanatory that they can still override these values if they need to.

**Review by JD-Robertson — APPROVED** [review]

**Review by spanglerco — APPROVED** [review]

**Review by chotea — APPROVED** [review]

File: release-notes/2023-03/README.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
+# SystemLink Enterprise 2023-03 Release Notes
+
+The 2023-03 bundle for SystemLink Enterprise has been published to <https://niedge01.jfrog.io>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.
+
+## New Features and Behavior changes
+
+- Support for Kubernetes 1.23
+
+- Support for both PostgreSQL 13 and 14.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:9]:
```suggestion
- Support for PostgreSQL 13 and 14
```

File: release-notes/2023-03/README.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+- Users can now export data tables to CSV
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:11]:
```suggestion
- You can export data tables to CSV
```

File: release-notes/2023-03/README.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+- Product, results, and steps tables can be queried using the table query builder.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:13]:
```suggestion
- You can query product, results, and steps tables with the table query builder
```

File: release-notes/2023-03/README.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+- OpenTelemetry metrics are now available for the following services
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:15]:
```suggestion
- OpenTelemetry metrics are available for the following services
```

File: release-notes/2023-03/README.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+    - testmonitor
+    - dataframeservice
+    - Kafka
+
+- The endpoint to view claims for the current logged in user, `/oidc/userinfo`, has been removed and replaced with `oidc/user-info`.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:20]:
```suggestion
- `oidc/user-info` replaced `/oidc/userinfo` as the endpoint to view claims for the current logged in user
```

File: release-notes/2023-03/README.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+- Kafka UI is now available for debugging and monitoring
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:22]:
```suggestion
- Kafka UI is available for debugging and monitoring
```

File: release-notes/2023-03/README.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+
+- Kafka S3 sink connectors are automatically deleted if a data table is marked `endOfData`.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:24]:
```suggestion
- Kafka S3 sink connectors are automatically deleted if a data table is marked `endOfData`
```

File: release-notes/2023-03/README.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+
+- Deleted Kafka S3 sink connectors are automatically reestablished if connectors are manually deleted.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:26]:
```suggestion
- Deleted Kafka S3 sink connectors are automatically reestablished if connectors are manually deleted
```

File: release-notes/2023-03/README.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+
+- The default memory provided to Kafka topic operator has increased from 256MB to 512MB to increase the total number of tables that can be written to by the Data Frame Service.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:28]:
```suggestion
- The default memory provided to Kafka topic operator has increased from 256MB to 512MB to increase the total number of tables the Data Frame Service can write to
```

File: release-notes/2023-03/README.md (hunk 3 above)
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:35]:
```suggestion
        - You must specify the single required image pull secret in a new value `global.niImagePullSecret` and reference this value in `global.imagePullSecrets`
```

File: release-notes/2023-03/README.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+        - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-03/getting-started/templates/systemlink-values.yaml#L18)
+
+    - Enabling Kafka OpenTelemetry metrics
+        - Set `global.telemetry.enabled` flag to `true`.
+        - You are no longer required to also set options in `kafka.metricsConfigs`, `kafka.jmxOptions`, and `kafka.kafkaExporter` configurations.
+        - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-03/getting-started/templates/systemlink-values.yaml#L56)
+
+    - Enabling user claims to be returned in a web browser.
+        - Set `webserver.oidc.enableUserInfo` to `true`.
+        - You should remove `offline_access` from `webserver.oidc.scope`.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:45]:
```suggestion
        - Remove `offline_access` from `webserver.oidc.scope`.
```

File: release-notes/2023-03/README.md [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+        - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-03/getting-started/templates/systemlink-values.yaml#L110).
+
+    - Enabling Kafka UI
+        - To enable the Kafka UI you must override the image pull repository to the repository you are installing SystemLink Enterprise from.
```
Comment (MelissaHilliard) [inline release-notes/2023-03/README.md:49]:
```suggestion
        - To enable the Kafka UI, you must override the image pull repository and specify the repository you are installing SystemLink Enterprise from.
```

File: release-notes/2023-03/README.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+        - Set `global.telemetry.enabled` flag to `true`.
+        - You are no longer required to also set options in `kafka.metricsConfigs`, `kafka.jmxOptions`, and `kafka.kafkaExporter` configurations.
+        - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-03/getting-started/templates/systemlink-values.yaml#L56)
+
+    - Enabling user claims to be returned in a web browser.
+        - Set `webserver.oidc.enableUserInfo` to `true`.
+        - You should remove `offline_access` from `webserver.oidc.scope`.
+        - [View this configuration](https://github.com/ni/install-systemlink-enterprise/blob/2023-03/getting-started/templates/systemlink-values.yaml#L110).
+
+    - Enabling Kafka UI
```
Comment (cameronwaterman) [inline release-notes/2023-03/README.md:48]: This section is no longer needed after the following PR goes in:
https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/454125

Once we rebuild the release bundle and validate this I will update the template values file and we should be able to remove this section from the docs.

Comment (cameronwaterman) [reply release-notes/2023-03/README.md:48]: Well, that's not exactly true. We will no longer need to specify the image registry override. We should add a blurb here about needing the specify the image pull secrets as outlined by the following PR:

https://github.com/ni/install-systemlink-enterprise/pull/116

File: release-notes/2023-03/README.md [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+
+- Kafka S3 sink connectors are automatically deleted if a data table is marked `endOfData`.
+
+- Deleted Kafka S3 sink connectors are automatically reestablished if connectors are manually deleted.
+
+- The default memory provided to Kafka topic operator has increased from 256MB to 512MB to increase the total number of tables that can be written to by the Data Frame Service.
+
+## Helm Chart Breaking Changes
+
+- `systemlink 0.11.81`
```
Comment (cameronwaterman) [inline release-notes/2023-03/README.md:32]: This has been rebuilt.

```suggestion
- `systemlink 0.11.85`
```

The only change in this release bundle is a new version of the dataframeservice chart (0.8.123), which pulled in a new version of the kafka-ui chart (0.5.4).

https://dev.azure.com/ni/DevCentral/_build/results?buildId=4433740&view=logs&j=c139e765-cd1b-5f1c-f29e-7c4d3e9388b9&t=eca95393-96ed-5496-e1f5-f82b94ffef11

File: release-notes/2023-03/README.md (hunk 15 above)
Comment (prestwick) [reply release-notes/2023-03/README.md:48]: @cameronwaterman please check the wording for this section. A couple of questions that might have me change the wording as it stands now:
- Would this Kafka-ui image pull secret be the same as the pull secrets specified in other places?
- Is there any update to the secrets template that would need to be required, or would we simply reference the pull-secret value as we would in other places?

Comment (cameronwaterman) [reply release-notes/2023-03/README.md:48]: We might want to be a bit more specific. Maybe the section header should read "Enabling Kafka UI for the Data Frame service Kafka deployment" (or something along those lines), as the Kafka cluster we deploy is specific to the DFS and deployed by the DFS.

And then the details would read "To enable Kafka UI for the Data Frame service Kafka deployment, specify `dataframeservice.kafka-ui.imagePullSecrets`. We might want to also link to the [use of the YAML alias in the template values](https://github.com/ni/install-systemlink-enterprise/blob/main/getting-started/templates/systemlink-values.yaml#L613) to drive home the point that this value can and should be single sourced.

Yes the image pull secret is the same and there is no required updates to the secrets template.

**Pull request merged**